File: SOEN3014 工程项目实践（卓越工程师型）/课程作业/大数据检索速度竞赛/代码（Spark）/src/main/scala/DatasetScale.scala
enum DatasetScale(val ACCOUNT_NUM: Long,
                  val FILE_NUM: Int,
                  val DATA_FOLDER_A: String,
                  val DATA_FOLDER_B: String):

  case BIG extends DatasetScale(
    5000 * 10000 * 10L,
    200,
    "data/account_db_big_a",
    "data/account_db_big_b"
  )

  case SMALL extends DatasetScale(
    BIG.ACCOUNT_NUM / 10,
    BIG.FILE_NUM / 10,
    "data/account_db_small_a",
    "data/account_db_small_b"
  )

  case TINY extends DatasetScale(
    BIG.ACCOUNT_NUM / 100,
    BIG.FILE_NUM / 100,
    "data/account_db_tiny_a",
    "data/account_db_tiny_b"
  )

File: SOEN3014 工程项目实践（卓越工程师型）/课程作业/大数据检索速度竞赛/代码（Spark）/src/main/scala/dataset_scale.py
"""A module that holds the scale of the dataset.
"""

from dataclasses import dataclass
from enum import Enum


class DatasetScale(Enum):
    """An enum that holds the scale of the dataset.
    """

    @dataclass
    class DatasetScale:
        """A dataclass that holds the scale of the dataset.
        """
        account_num: int
        file_num: int
        data_folder_a: int
        data_folder_b: int

    BIG = DatasetScale(
        5000 * 10000 * 10,
        200,
        'data/account_db_big_a',
        'data/account_db_big_b'
    )

    SMALL = DatasetScale(
        5000 * 10000,
        20,
        'data/account_db_small_a',
        'data/account_db_small_b'
    )

    TINY = DatasetScale(
        5000 * 1000,
        2,
        'data/account_db_tiny_a',
        'data/account_db_tiny_b'
    )
